mesheryctl/pkg/utils/helpers.go

```diff
@@ -158,18 +158,55 @@ func SetFileLocation() error {
 
 //PreReqCheck prerequisites check
 func PreReqCheck() error {
+	//Check whether docker daemon is running or not
+	if err := exec.Command("docker", "ps").Run(); err != nil {
+		log.Info("Docker daemon is not running")
+		//No auto installation of docker for windows
+		if runtime.GOOS == "windows" {
+			return errors.Wrap(err, "please start Docker daemon. Run `mesheryctl system start` once docker is started.")
+		}
+		err = startdockerdaemon()
+		if err != nil {
+			return errors.Wrap(err, "failed to start Docker daemon. Run `mesheryctl system start` once docker is started.")
+		}
+	}
 	//Check for installed docker-compose on client system
 	if err := exec.Command("docker-compose", "-v").Run(); err != nil {
 		log.Info("Docker-Compose is not installed")
 		//No auto installation of Docker-compose for windows
 		if runtime.GOOS == "windows" {
-			return errors.Wrap(err, "please install docker-compose")
+			return errors.Wrap(err, "please install docker-compose. Run `mesheryctl system start` after docker-compose is installed.")
 		}
 		err = installprereq()
 		if err != nil {
-			return errors.Wrap(err, "failed to install prerequisites")
+			return errors.Wrap(err, "failed to install prerequisites. Run `mesheryctl system start` after docker-compose is installed.")
+		}
+	}
+	return nil
+}
+
+func startdockerdaemon() error {
+	log.Info("Attempting to start Docker daemon")
+	// TODO: read user input on whether to start docker or not.
+	// once user gaves permission, start docker daemon on linux/macOS
+	if runtime.GOOS == "linux" {
+		if err := exec.Command("service", "docker", "start").Run(); err != nil {
+			return errors.Wrap(err, "please start Docker then run the command `mesheryctl system start` to start Meshery")
+		}
+	} else {
+		// Assuming we are on macOS, try to start Docker from default path
+		cmd := exec.Command("/Applications/Docker.app/Contents/MacOS/Docker")
```

Comment:
Minor Question: If docker is not in this path could we get the path from the user ? or run a sort of which to find it

Reply:
This is best effort attempt, if the command fails to find Docker application in the given path, user will see the informational message to reattempt "mesheryctl system start" once docker is started. I have also added a check to see if docker daemon gets started after the sleep time, if not, an info message is displayed to user.